fog/view/enclave/impl/src/lib.rs

```diff
@@ -352,10 +360,31 @@ where
         )
     }
 
+    // Takes each MultiViewStoreResponseQuery
     fn get_minimum_highest_processed_block_count(
-        &self,
-        _shard_query_responses: Vec<(QueryResponse, BlockRange)>,
+        responses: &[DecryptedMultiViewStoreQueryResponse],
     ) -> u64 {
-        todo!()
+        let responses_with_gaps = responses
+            .iter()
+            .cloned()
+            .filter(|response| {
+                response.query_response.highest_processed_block_count
+                    != response.block_range.end_block
```

Comment:
:wrench: Since `end_block` is exclusive should this be something like
```suggestion
                response.query_response.highest_processed_block_count + 1
                    != response.block_range.end_block
```

Looking around some tests, I don't think so, but I think there may need to be some better disambiguation in the descriptions (follow on PR/issue if agreed)

Reply:
I changed the logic that calculates this, but the underlying idea is the same. If the block_count is less than the end range block index (which is non inclusive), the store hasn't finished processing its blocks.